### ISSUE TITLE
Talkback borders not drawn with new Add2App embedding using RenderMode.surface and TransparencyMode.opaque on certain Android devices

### ISSUE THREAD

**Issue by dannyvalentesonos** [flutter/flutter#53889]
When using Flutter in an Android Add2App scenario, building the flutter fragment like so:

```
flutterFragment = FlutterFragment.withCachedEngine(FlutterDelegate.getInstance().getFlutterEngineId())
                    .renderMode(RenderMode.surface)
                    .transparencyMode(TransparencyMode.opaque)
                    .build();
```

It is my understanding from #37025 that Talkback borders should work. Also, reading the documentation for RenderMode, it sounds like the surface when opaque should be placed on the bottom of the surface's window, so TalkBack borders should work.

```
public enum TransparencyMode {
/**
   * Renders a Flutter UI without any transparency. This affects Flutter UI's with {@link
   * RenderMode#surface} by introducing a base color of black, and places the {@link
   * FlutterSurfaceView}'s {@code Window} behind all other content.
   *
   * <p>In {@link RenderMode#surface}, this mode is the most performant and is a good choice for
   * fullscreen Flutter UIs that will not undergo {@code Fragment} transactions. If this mode is
   * used within a {@code Fragment}, and that {@code Fragment} is replaced by another one, a brief
   * black flicker may be visible during the switch.
   */
  opaque,
```

Finally, also something interesting, is that if the device's screen locks, when unlocking it, talkback borders suddenly start to be displayed. 

I feel that using a Surface in opaque cases should work with Talkback as it wouldn't be great if we had to use the Texture mode just for Talkback, as it's recommended not to for performance reasons.

The device I'm testing on is a Huawei P30.

**Comment by dannyvalentesonos** [comment]
flutter doctor -v
[✓] Flutter (Channel dev, v1.16.2, on Mac OS X 10.14.6 18G4032, locale en-US)
    • Flutter version 1.16.2 at /Users/danny.valente/Library/flutter/1.16.2-dev
    • Framework revision 58cad787d8 (10 days ago), 2020-03-23 08:33:41 -0700
    • Engine revision 2d42c74a34
    • Dart version 2.8.0 (build 2.8.0-dev.16.0 ba8baa46b4)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/danny.valente/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C505
    ! CocoaPods 1.7.5 out of date (1.8.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade:
        sudo gem install cocoapods

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 44.0.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

**Comment by dannyvalentesonos** [comment]
@dnfield Here's the new issue. Thanks!

**Comment by mklim** [comment]
Thanks for filing this @dannyvalentesonos. I'm not going to be able to meaningfully debug this in the immediate future, but for anyone who does one thing to look into is whether or not `setZOrderOnTop` is called somehow in this code path. As far as I know it _shouldn't_ be, but if it were still somehow being called the surface would end up covering the Talkback borders like in #37025.

**Comment by dannyvalentesonos** [comment]
@mklim Thanks for the response. I actually debugged this and it doesn't look like `setZOrderOnTop` is called at all when using TransparencyMode.opaque, as far as I can tell.

So I decided to test with a Nexus 9, and interestingly, it works fine on the Nexus 9. So this bug seems like it is a certain set of devices. I only have the Huawei P30 and the Nexus 9 to test with, so I can't verify on other devices at the moment, but I will update here if and when I find out more.

**Comment by dannyvalentesonos** [comment]
We have done more testing. We have tested on the following devices:

- Google Nexus 9
- Google Nexus 10
- Google Pixel 1
- Samsung Tab A (2017)
- Samsung Galaxy S9+
- Samsung Galaxy S8
- Huawei Honor 8
- Huawei Mate 20
- Huawei P30
- Huawei MediaPad M5 8.4
- Huawei MediaPad M5 10.8

and all Huawei devices presented this problem, while the Google and Samsung devices behaved correctly.

**Comment by mklim** [comment]
Glad you figured this out, thanks for the update!
